pkg/cloud/api/path.go

```diff
@@ -92,6 +147,20 @@ func (p Path) HasPrefix(prefix Path) bool {
 	return true
 }
 
+// isSliceIndex returns true if the element is a SliceIndex
+func isSliceIndex(element string) bool {
+	if len(element) < 3 && element[0] != pathSliceIndex {
```

Comment:
why 3? isn't !1 a valid index?

Reply:
fixed, sorry for bug, added unit tests

Reply:
ok      github.com/GoogleCloudPlatform/k8s-cloud-provider/pkg/cloud/api (cached)        coverage: 85.9% of statements